src/Symfony/Component/DependencyInjection/Compiler/RemoveUnusedDefinitionsPass.php

```diff
@@ -59,7 +59,7 @@ public function process(ContainerBuilder $container)
                         $referencingAliases[] = $node->getValue();
                     }
                 }
-                $isReferenced = (count(array_unique($sourceIds)) - count($referencingAliases)) > 0;
+                $isReferenced = (count(array_count_values($sourceIds)) - count($referencingAliases)) > 0;
```

Comment:
I find the code `count(array_unique($ids))` much more readable about its intent.

As this code runs at warmup time, and difference will be void on PHP 7.2 (thanks to the optimization done by @nikic), I suggest keeping the existing code.